File: module-services/src/main/java/eu/geekhome/services/configurable/RequiredLongValidator.java
package eu.geekhome.services.configurable;


import eu.geekhome.services.localization.Resource;

public class RequiredLongValidator implements Validator<Long> {
    @Override
    public Resource getReason() {
        return new Resource("This field is required",
                "To pole jest wymagane");
    }

    @Override
    public boolean validate(Long fieldValue) {
        return fieldValue != null && fieldValue > 0;
    }
}

File: module-domain/src/main/java/eu/geekhome/services/configurable/RequiredLongValidator.kt
package eu.geekhome.services.configurable

import eu.geekhome.services.R
import eu.geekhome.services.localization.Resource

class RequiredLongValidator : Validator<Long> {

    override fun validate(fieldValue: Long?): Boolean {
        if (fieldValue == null) {
            return false
        }

        return fieldValue > 0
    }

    override val reason: Resource = R.validator_required_field
}